src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -920,6 +920,11 @@ private void joinTokenRing(int delay, boolean autoBootstrap, Collection<String>
                 }
                 Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
             }
+            while (!isNewCluster() && Schema.instance.getVersion().equals(Schema.emptyVersion))
```

Comment:
should we do the `!isNewCluster()` in a wrapping if statement seeing as how it will never update?